apps/passport-client/components/screens/DeviceLoginScreen.tsx

```diff
@@ -0,0 +1,82 @@
+import { Spacer } from "@pcd/passport-ui";
+import {
+  ChangeEvent,
+  FormEvent,
+  useCallback,
+  useContext,
+  useState
+} from "react";
+import styled from "styled-components";
+import { DispatchContext } from "../../src/dispatch";
+import { BigInput, Button, CenterColumn, H2, TextCenter } from "../core";
+import { AppContainer } from "../shared/AppContainer";
+
+export function DeviceLoginScreen() {
+  const [email, setEmail] = useState("");
+  const [secret, setSecret] = useState("");
+  const [state, dispatch] = useContext(DispatchContext);
+
+  const onAuthenticate = useCallback(
+    function (e: FormEvent<HTMLFormElement>) {
+      e.preventDefault();
+      dispatch({
+        type: "new-device-login-passport"
+      });
+      dispatch({
+        type: "device-login",
+        secret: secret,
+        email: email
+      });
+    },
+    [dispatch, email, secret]
+  );
```

Comment:
nit: we generally use arrow functions when possible; another thing is if you make make this function be called on button click rather than on form submit you'll be able to remove this `e.preventDefault()`

Reply:
I copied both of those things from the `onGenPass` function in `LoginScreen.tsx`!